core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -414,11 +429,91 @@ object ContractInfoV0TLV extends TLVFactory[ContractInfoV0TLV] {
   }
 }
 
+case class TLVPoint(outcome: Long, value: Satoshis, isEndpoint: Boolean)
+    extends NetworkElement {
+
+  override def bytes: ByteVector = {
+    val leadingByte = if (isEndpoint) {
+      1.toByte
+    } else {
+      0.toByte
+    }
+
+    ByteVector(leadingByte) ++ BigSizeUInt(outcome).bytes ++ UInt64(
+      value.toLong).bytes
+  }
+}
+
+object TLVPoint extends Factory[TLVPoint] {
+
+  override def fromBytes(bytes: ByteVector): TLVPoint = {
+    val isEndpoint = bytes.head match {
+      case 0 => false
+      case 1 => true
+      case b: Byte =>
+        throw new IllegalArgumentException(
+          s"Did not recognize leading byte: $b")
+    }
+
+    val outcome = BigSizeUInt(bytes.tail)
+    val value = UInt64(bytes.drop(1 + outcome.byteSize).take(8))
+    TLVPoint(outcome.toLong, Satoshis(value.toLong), isEndpoint)
+  }
+}
+
+case class ContractInfoV1TLV(
+    base: Int,
+    numDigits: Int,
+    totalCollateral: Satoshis,
+    points: Vector[TLVPoint])
+    extends ContractInfoTLV {
+  override val tpe: BigSizeUInt = ContractInfoV1TLV.tpe
+
+  override val value: ByteVector = {
+    BigSizeUInt(base).bytes ++ UInt16(numDigits).bytes ++ UInt64(
+      totalCollateral.toLong).bytes ++ BigSizeUInt(
+      points.length).bytes ++ points.foldLeft(ByteVector.empty)(_ ++ _.bytes)
+  }
+}
+
+object ContractInfoV1TLV extends TLVFactory[ContractInfoV1TLV] {
```

Comment:
imo this should be named `LargeRangeContractInfoV0TLV` as this one doesn't fully replace contract info v0